src/components/LabEnvironment/LabEnvironment.tsx

```diff
@@ -61,6 +75,59 @@ export class LabEnvironment extends Component<LabEnvironmentProps, LabEnvironmen
     return this.props.userLab.status !== 'Started';
   }
 
+  setLabEndDateTimeInterval = (userLab: UserLab) => {
+    this.state.interval = window.setInterval(() => {
+      this.setState({ labEndDateTime: userLab.endDateTime! });
+    }, 1000);
+  }
+
+  handleExtendEndDateTime = async () => {
+    const updatedUserLab = await updateEndDateTime(this.props.userLab.id);
+    window.clearInterval(this.state.interval);
+    this.setState({ labEndDateTime: updatedUserLab.endDateTime! });
+    this.setLabEndDateTimeInterval(updatedUserLab);
+  };
+
+  showExtendButton = () => {
+    const remainingLabTime = getLuxonObjectFromString(this.state.labEndDateTime!).toLocal().diffNow().as('second');
+    // shows a pop up if the lab times up instead of providing an extend option
+    if (getRemainingLabTime(this.state.labEndDateTime) === 'Times up!') {
+      window.setTimeout(() => {
+        window.location.replace(RoutePaths.myModules);
+      }, 15000);
+      return (
+        <Layout>
```

Comment:
The pop-up should be its own component,
Also can you add a break before the redirect sentence and add italicization and make it one h value smaller?

Reply:
Do you mean the pop-up should be in a separate file or just a separate component inside the same file. If we do it in a separate file, it still might not be very re-usable since we still have to pass the values and text, and this is already using react bootstrap's Modal component.

Reply:
My thought was to put it in its own file, then pass a title, body., and the redirect, but I could be overthinking it.
@jasekiw thoughts?

Reply:
I talked to John about it, he had pop-up in the same file in #67, I think Jason ok'd it since it got merged.

Reply:
I trust you, I will approve this.

Reply:
@zkhussain I think this file is large enough, putting the component in it's own file makes the most sense here.